### PR TITLE
gifload: ensure total height of all pages is sanitised

### PR DESCRIPTION
**Pull request by lovell**
Helps prevent an integer overflow, plus adds a >60k page GIF sample to aid future fuzz testing.

We should probably backport this to 8.10 also.

### CODE REVIEW

File: libvips/foreign/gifload.c [hunk 1]
```diff
@@ -700,8 +700,12 @@ vips_foreign_load_gif_scan_extension( VipsForeignLoadGif *gif )
 static int
 vips_foreign_load_gif_set_header( VipsForeignLoadGif *gif, VipsImage *image )
 {
+	const int total_height = gif->file->SHeight * gif->n;
+	if ( total_height <= 0 || total_height > VIPS_MAX_COORD ) {
```
Comment (jcupitt) [inline libvips/foreign/gifload.c:704]: You could probably still get int overflow here. How about doing the multiplication as gint64? That would be safe.

Comment (jcupitt) [reply libvips/foreign/gifload.c:704]: You need to cast one of the args to the * as well, or the mult will be done as int :(

Comment (lovell) [reply libvips/foreign/gifload.c:704]: Thanks, I've updated this, plus added an appropriate message to the error log.

**Comment by jcupitt** [comment]
Great!

**Comment by jcupitt** [comment]
I backported to 8.10 too.

**Pull request merged**